app/uk/gov/hmrc/exports/movements/config/AppConfig.scala

```
@@ -38,8 +38,8 @@ class AppConfig @Inject()(runModeConfiguration: Configuration, servicesConfig: S
 
   def clientIdInventory(implicit hc: HeaderCarrier): String = {
     val userAgent = hc.headers.find(_._1.toLowerCase() == "user-agent").map(_._2).getOrElse {
-      logger.warn("Request had missing User-Agent header. Falling Back to a default Client ID")
-      "default"
+      logger.warn("Request had missing User-Agent header.")
+      throw MissingClientIDException("User Agent")
```

I don't think you technically need to throw that extra exception.  If you returned an empty string the following line will throw a similar exception.  However, this is ok too.

You could probably use "getOrThrow" (instead of getOrElse) and skip the logging of the warning - no point logging of you are going to stop the app starting with the exception.

---

I help Alsahid with this implementation. If you use empty string, this empty string will be putted in the configuration and application will try to retrieve the incorrect key. We need to throw an exception, because this case is incorrect

---

I think it would be good to log the service name we can't find client-id for.